src/main/java/com/suse/saltstack/netapi/parser/SaltStackKeyParser.java

```diff
@@ -0,0 +1,66 @@
+package com.suse.saltstack.netapi.parser;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+import com.suse.saltstack.netapi.client.SaltStackKeyResult;
+import com.suse.saltstack.netapi.exception.SaltStackParsingException;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.util.Map;
+
+/**
+ * Implements a parser for public key fingerprints (from endpoint /keys)
+ */
+public class SaltStackKeyParser implements ISaltStackResultParser {
+
+    private SaltStackKeyResult keyResult = new SaltStackKeyResult();
+
+
+    @Override
+    public <T> T parse(Type resultType, InputStream inputStream) throws SaltStackParsingException {
+        // uncomment this on Java 8
+        //if ( !keyResult.getClass().getTypeName()
+        //        .equals(resultType.getTypeName())) {
+        if ( !"class com.suse.saltstack.netapi.client.SaltStackKeyResult"
```

Comment:
Some other points:

* in general, please make sure your code compiles and works as intended in Java 7 for this project, as we are not sure if we will be able to support Java 8 yet. Sorry not to have mentioned that upfront, hopefully it's not too much work to adapt code to the older language;
* second, can you propose some alternative design in which reflection is not used to check types/raise exceptions at runtime? I guess for example that having a parametric `ISaltStackResultParser` interface would make more sense than a parametric `parse` method, as each concrete implementation would only handle a specific type.

Thanks!